src/ReleasableImage.cpp

```diff
@@ -0,0 +1,34 @@
+#include <ZividPython/ReleasableImage.h>
+
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+namespace
+{
+    py::buffer_info makeBufferInfo(ZividPython::ReleasableImage &image)
+    {
+        const auto data = image.dataPtr();
+
+        return py::buffer_info{ const_cast<void *>(static_cast<const void *>(data)), // HACKY HACK WATCH OUT
```

Comment:
We do not have a non const dataPtr for the 2d image class, so we need to hack around this by casting away to const. This should ideally be done in another way, perhaps @nedrebo has some ideas how to do this properly.

Reply:
I don't think we need a hack here, but if we do we need a nice description about why.

However I would surprised if pybind11 did not support making numpy views out of const memory areas. Did you look for that? If not it is actual worth a bug report, but I would look thoroughly first, because this is something that I think is logical to have in this library.

Reply:
I'll hold back the Approve until it's either improved or the comment line is made more helpful/descriptive.

Reply:
Sounds good, will look around for a fix.

Reply:
I looked at the source for pybind11, no support for const void *, but a flag for read only was added here: https://github.com/pybind/pybind11/pull/1466

I would use that and also change the cast to const_cast to be explicit about what is going on. Cast to void is not necessary, so this should be enough: `const_cast<PixelFormat *>(data)`.

Reply:
Updated the code and comment